src/internal/connector/onedrive/collection.go

```diff
@@ -390,6 +375,42 @@ func (oc *Collection) getDriveItemContent(
 	return itemData, nil
 }
 
+// downloadContent attempts to fetch the item content.  If the content url
+// is expired (ie, returns a 401), it re-fetches the item to get a new download
+// url and tries again.
+func downloadContent(
+	ctx context.Context,
+	svc graph.Servicer,
+	igf itemGetterFunc,
+	irf itemReaderFunc,
+	gr graph.Requester,
+	item models.DriveItemable,
+	driveID string,
+) (io.ReadCloser, error) {
+	_, content, err := irf(ctx, gr, item)
+	if err == nil {
+		return content, nil
+	} else if !graph.IsErrUnauthorized(err) {
+		return nil, err
+	}
+
+	// Assume unauthorized requests are a sign of an expired jwt
+	// token, and that we've overrun the available window to
+	// download the actual file.  Re-downloading the item will
+	// refresh that download url.
+	di, err := igf(ctx, svc, driveID, ptr.Val(item.GetId()))
+	if err != nil {
+		return nil, clues.Wrap(err, "retrieving expired item")
+	}
+
+	_, content, err = irf(ctx, gr, di)
```

Comment:
Previously we used to try thrice to download expired content. Should we retain that behaviour instead of trying once.

Reply:
The middleware does retry things, but it does not retry link expiring(and retrying using new link).

Reply:
I don't believe the link can expire more than once.  Since we regenerate and consume the link at the same time, the second one won't have any delay before its usage.